dbms/src/Functions/FunctionsRound.h

```diff
@@ -1107,6 +1114,33 @@ struct TiDBIntegerRound
             return castBack((input < 0), absolute_value);
         }
     }
+
+    static OutputType evalTruncate(InputType input, FracType frac)
+    {
+        // modified from https://github.com/pingcap/tidb/blob/807b8923c0181d89d4ea8e4195f9d27d299298a7/pkg/expression/builtin_math.go#L2196-L2219
+        const auto value = static_cast<OutputType>(input);
+        if (frac >= 0)
+            return value;
+        else if (frac <= -max_digits)
+            return 0;
+        else
+        {
+            // To make sure static_cast<OutputType>(Pow::result[-frac]) will not overflow.
+            assert(Pow::result[max_digits] < std::numeric_limits<OutputType>::max());
```

Comment:
Is it Pow::result[-frac] instead of max_digits here? And since frac > -max_digits, there seems no need to check here? Or at least no need to assert the program here?

Reply:
changed.